feat(dx): add /lfx-pr slash command for PR shipping (#672)
linuxfoundation/lfx-self-serve: .claude/commands/lfx-pr.md

---

> description: Ship a PR for lfx-v2-ui — runs quality gates, signs commits (DCO+GPG), pushes branch, opens PR with LFXV2 linkage.

Author: Fixed — updated the frontmatter description to `(DCO sign-off; GPG optional)` so it matches the command's actual behavior. Step 2 still recommends `-S` for users who have GPG configured but no longer enforces it.

See commit c713ead9.

argument-hint: [optional: "<commit subject>" — if working tree has uncommitted changes]
---

# /lfx-pr — Ship a PR for lfx-v2-ui

You are running in the lfx-v2-ui monorepo. The user is invoking this command to ship the current branch as a PR. Follow these steps in order. Fail fast — stop immediately if any step fails and surface the error.

User-provided argument (if any): $ARGUMENTS

## Step 1: Pre-flight state checks

Run these in order. If any fail, stop and report:

- git rev-parse --abbrev-ref HEAD → if "main", stop: "Refusing to PR from main."

> - git rev-list --count main..HEAD → if 0, stop: "No commits ahead of main."

Author: Done — added `git fetch origin main` to Step 1, switched the ahead check to `origin/main..HEAD`, and updated the Step 4 self-review diff to `git diff origin/main...HEAD`. Brings the command in line with how `.claude/skills/preflight` already does its comparisons. Stops the failure mode where a stale local main makes the ahead count wrong or the self-review diff miss recent base changes.

See commit c713ead9.

- git status --porcelain

If git status --porcelain is non-empty:

- If user passed an argument, treat it as commit subject and proceed to Step 2
- If no argument, stop and ask for one

If working tree is clean, skip Step 2.

## Step 2: Commit (only if dirty + subject provided)

Validate subject:

- Format: <type>(<scope>): <subject> — Angular convention
- Allowed types: feat, fix, refactor, perf, test, docs, style, build, ci, revert
- Banned: chore:
- Length: ≤72 chars
- Scope: lowercase module name or shared/bff/ssr

> If invalid, stop and report. Otherwise:

Author: Good catch — the scope rule was narrower than `.claude/rules/commit-workflow.md` actually allows. Relaxed it to "lowercase, describes the affected area (e.g., auth, ui, api, docs, ci, dx)" with a pointer to the rules file, and added a follow-up line delegating final validation to `yarn commitlint` (which the commit-msg hook runs anyway). That avoids the doc maintaining its own divergent allowlist.

See commit c713ead9.

git add -A

> # Note: -S (GPG signing) is recommended but not enforced by repo policy. Add it if you have GPG configured.

Author: Done — converted the `# Note:` H1-style line to a Markdown blockquote (`> **Note:** ...`) and removed the trailing period. Also wrapped the surrounding `git add` / `git commit` commands in a fenced bash block so they render as code rather than bare paragraphs (Prettier had been flattening the indentation in the previous version, which is what created the H1 in the first place).

See commit c713ead9.

git commit --signoff -m "<subject>"

--signoff is mandatory (DCO). -S adds GPG signing — recommended but not enforced by repo policy. If --signoff fails, stop.

## Step 3: Quality gates (fail-fast)

Run sequentially, stop on first failure, surface output:
yarn check-types
yarn lint
yarn format
yarn test
yarn build

> Do not auto-fix beyond what --fix flags do. Do not skip yarn build (catches SSR breaks).

Author: Done — switched Step 3 from `yarn lint` / `yarn format` (auto-fix) to `yarn lint:check` / `yarn format:check` (read-only). Gates no longer mutate the working tree at ship time. If lint or format reports issues, the command now tells the user to fix them, return to Step 2 to commit the fixes, then re-run gates. That keeps the "ship a PR" command honest — no silent auto-fixes pushed without a commit.

See commit c713ead9.

## Step 4: Self-review

git diff main...HEAD — read it. Check against CLAUDE.md and .claude/rules/:

- Consider: any browser-only APIs (window, document, etc.) — if found outside isPlatformBrowser guards, flag for the user to confirm (SSR risk)
- Consider: hard-coded brand hex values — if found, suggest using lfxColors scales instead (avoids drift when LF brand updates)
- Module scope respected
- No nested ternaries
- No interface Foo {} inside apps/
- License headers on new files
- Commit messages: Angular format, ≤72, no chore:
- Every commit has Signed-off-by trailer

> These are advisory checks — surface findings to the user but do not block. The user decides whether to fix before pushing.

Author: Good point, and you're correct that commit-format, ≤72 chars, no-`chore:`, and `Signed-off-by` are all CI/hook-enforced (commitlint + DCO check). Making them advisory in a "ship a PR" command is inconsistent with the fail-fast posture elsewhere.

This conflicts with an explicit design decision earlier in the doc's history where Step 4 was reworded to be advisory ("surface findings to the user but do not block"). That intent applies cleanly to the taste items (isPlatformBrowser, lfxColors, ternaries, no-interface-in-apps), but not to the hard-enforced ones.

Leaving this thread open pending a decision on whether to split Step 4 into a hard fail-fast block (commit format, ≤72, no chore:, Signed-off-by, license headers) and an advisory block (taste items). Will follow up in a separate commit once that's decided.

> These are advisory checks — surface findings to the user but do not block. The user decides whether to fix before pushing.

Author: Accepted @copilot-swe-agent's autofix (6537f5f1) as-is — it implements the hard fail-fast / advisory split cleanly and matches the option we previously discussed. Step 4 now runs `./check-headers.sh`, `yarn commitlint --from origin/main --to HEAD`, and a Signed-off-by + GPG-signature trailer check up front (fail-fast), with taste items (isPlatformBrowser, lfxColors, ternaries, etc.) in a separate non-blocking block below.

Also added GPG-signature verification to the hard block in 576a788d (`git log --format='%G?' origin/main..HEAD` — every line must be `G`) to align with @jordane's clarification on the GPG-enforcement thread.

---
description: Ship a PR for lfx-v2-ui — runs quality gates, signs commits (DCO sign-off; GPG optional), pushes branch, opens PR with LFXV2 linkage.
argument-hint: [optional: "<commit subject>" — if working tree has uncommitted changes]
---

# /lfx-pr — Ship a PR for lfx-v2-ui

You are running in the lfx-v2-ui monorepo. The user is invoking this command to ship the current branch as a PR. Follow these steps in order. Fail fast — stop immediately if any step fails and surface the error.

User-provided argument (if any): $ARGUMENTS

## Step 1: Pre-flight state checks

Run these in order. If any fail, stop and report:

- git fetch origin main → ensure `origin/main` is current (avoids stale-local-main drift)
- git rev-parse --abbrev-ref HEAD → if "main", stop: "Refusing to PR from main."
- git rev-list --count origin/main..HEAD → if 0, stop: "No commits ahead of origin/main."
- git status --porcelain

If git status --porcelain is non-empty:

> - If user passed an argument, treat it as commit subject and proceed to Step 2

Reviewer: I don't think most users will think to put a commit message/subject at the end of this skill name, they'll likely instead pass extra instructions they want to modify things slightly. I.e `/lfx-pr move commits to a new branch jme/branch-name and start a pr`

Author: Good call — you're right that the "argument == commit subject" framing is wrong for how this kind of slash command actually gets used. Reworked it:

- **Frontmatter `argument-hint`**: `"<commit subject>"` → `any extra instructions — commit subject, target branch, draft mode, extra PR body context`
- **Body intro (after `$ARGUMENTS` reference)**: added a paragraph instructing the command to interpret `$ARGUMENTS` as freeform user instructions and apply each directive at the step that owns it (commit subject in Step 2, branch ops in Steps 1/5, draft flag in Step 6). If anything is ambiguous, ask the user before acting.
- **Step 1 dirty-tree branch**: previously "if arg → commit subject, else stop and ask". Now: if `$ARGUMENTS` includes or clearly implies a commit subject, use it; otherwise propose one from the staged diff and ask the user to confirm or revise.
- **Step 2 heading**: `Commit (only if dirty + subject provided)` → `Commit (only if dirty)` since the command can now propose a subject when one isn't supplied.

Your example (`/lfx-pr move commits to a new branch jme/branch-name and start a pr`) should now route correctly: the branch directive is honored in Steps 1/5, no commit subject was given so the command proposes one from the diff and confirms.

See commit 576a788d.

- If no argument, stop and ask for one

If working tree is clean, skip Step 2.

## Step 2: Commit (only if dirty + subject provided)

Validate subject:

- Format: <type>(<scope>): <subject> — Angular convention
- Allowed types: feat, fix, refactor, perf, test, docs, style, build, ci, revert
- Banned: chore:
- Length: ≤72 chars
- Scope: lowercase, describes the affected area (e.g., auth, ui, api, docs, ci, dx) — see `.claude/rules/commit-workflow.md`
- When in doubt, let `yarn commitlint` (run by commit-msg hook) be the source of truth

If invalid, stop and report. Otherwise:

```bash
git add -A
git commit --signoff -m "<subject>"
```

> > **Note:** `-S` (GPG signing) is recommended but not enforced by repo policy — add it if you have GPG configured

Reviewer: It is enforced, please change this.

Author: Thanks for the correction — fixed. The doc now treats `-S` (GPG signing) as mandatory alongside `--signoff` (DCO), and I propagated that everywhere it was previously framed as optional:

- Frontmatter description: `(DCO sign-off; GPG optional)` → `(DCO sign-off + GPG)`
- Step 2 commit command: restored `-S` flag
- Step 2 Note + mandatory sentence: both rewritten to state both flags are required
- Step 4 hard checks: added GPG-signature verification (`git log --format='%G?' origin/main..HEAD` — every line must be `G`) alongside the existing Signed-off-by trailer check
- Step 6 PR body checklist: line now reads "DCO sign-off (`--signoff`) and GPG signed (`-S`) on every commit"
- Failure-handling section: "missing DCO" → "missing DCO/GPG"

See commit 576a788d.

Side note: I couldn't directly verify the GPG-enforcement policy from outside (the branch protection API returned 404 for me — likely a permissions thing), so I took your word for it. If the policy doc is somewhere I can link to from `CLAUDE.md` or `.claude/rules/commit-workflow.md`, happy to add the reference.

---
description: Ship a PR for lfx-v2-ui — runs quality gates, signs commits (DCO sign-off + GPG), pushes branch, opens PR with LFXV2 linkage.
argument-hint: [optional: any extra instructions — commit subject, target branch, draft mode, extra PR body context]
---

# /lfx-pr — Ship a PR for lfx-v2-ui

You are running in the lfx-v2-ui monorepo. The user is invoking this command to ship the current branch as a PR. Follow these steps in order. Fail fast — stop immediately if any step fails and surface the error.

User-provided argument (if any): $ARGUMENTS

Interpret `$ARGUMENTS` as freeform user instructions. They may include any of: a commit subject (e.g., `feat(auth): add OAuth`), a target branch (e.g., `to feat/lfx-pr-v2`, `move commits to jme/branch-name`), a draft toggle (e.g., `as draft`), additional PR body context, or other adjustments to the default flow. Apply each instruction at the step that owns it (commit subject in Step 2, branch operations in Steps 1/5, draft flag in Step 6). If any directive is ambiguous, ask the user before acting on it.

> 

Author: Resolved by wiring `--draft` end-to-end rather than removing the feature.

A side-trip happened first: @copilot-swe-agent[bot] pushed `3d28b993` ("docs(dx): remove draft-mode refs from /lfx-pr args"), which took the *opposite* option — removing draft references from the args interpretation. That direction would have removed a useful feature without addressing the underlying intent. The user opted for the implement-end-to-end path, so I layered the wiring on top in `bbc00b46`:

- Frontmatter `argument-hint`: restored "draft mode" to the example list
- Args interpretation paragraph: restored "a draft toggle (e.g., `as draft`)" and the "draft flag in Step 6" routing hint
- Step 6 `gh pr create`: documented that a draft directive in `$ARGUMENTS` appends `--draft` to the `gh` invocation; manual-open fallback also notes the flag
- "When to use" section: replaced the "Not for drafts" exclusion with "Drafts — opt in via a draft directive in `$ARGUMENTS` (e.g., `as draft`)"

Default behavior is unchanged (review-ready PR). Drafts are explicit opt-in.

See commit bbc00b46.

- git rev-parse --abbrev-ref HEAD → if "main", stop: "Refusing to PR from main."
- git rev-list --count origin/main..HEAD → if 0, stop: "No commits ahead of origin/main."
- git status --porcelain

If git status --porcelain is non-empty:

- If `$ARGUMENTS` includes (or clearly implies) a commit subject — use it and proceed to Step 2.
- Otherwise, propose a commit subject derived from the staged diff and ask the user to confirm or revise before committing.

If working tree is clean, skip Step 2.

## Step 2: Commit (only if dirty)

Validate subject:

- Format: <type>(<scope>): <subject> — Angular convention
- Allowed types: feat, fix, refactor, perf, test, docs, style, build, ci, revert
- Banned: chore:
- Length: ≤72 chars
- Scope: lowercase, describes the affected area (e.g., auth, ui, api, docs, ci, dx) — see `.claude/rules/commit-workflow.md`
- When in doubt, let `yarn commitlint` (run by commit-msg hook) be the source of truth

If invalid, stop and report. Otherwise:

```bash
git add -A
git commit --signoff -S -m "<subject>"
```

> **Note:** Both `--signoff` (DCO) and `-S` (GPG signing) are required by repo policy. Configure GPG if you haven't already.

> Both `--signoff` (DCO) and `-S` (GPG signing) are mandatory per repo policy. If either fails, stop — do not push unsigned commits.

Author: Good eye on the apparent gap — but this one's already handled, just not in a way you can see from inside this PR.

@jordane (LF reviewer on this PR) explicitly confirmed in an earlier thread that GPG signing **is** enforced by repo policy. That's the authoritative source — the contributor-facing docs (`CLAUDE.md`, `.claude/rules/commit-workflow.md`) just hadn't caught up. PR #674 is in flight specifically to close that documentation gap:

- https://github.com/linuxfoundation/lfx-v2-ui/pull/674

Once #674 lands, future contributors won't hit the same confusion. Keeping `-S` as mandatory here matches @jordane's stated policy. Happy to revisit if @jordane updates the policy guidance.

## Step 3: Quality gates (fail-fast)

Run sequentially, stop on first failure, surface output:

```bash
yarn check-types
yarn lint:check
yarn format:check
yarn test
yarn build
```

Gates are read-only (`:check` variants) — they do not mutate the working tree. If lint or format reports issues, fix them, return to Step 2 to commit the fixes, then re-run gates. Do not skip `yarn build` (catches SSR breaks).

## Step 4: Self-review

Run these hard policy checks first (fail-fast):

```bash
./check-headers.sh
yarn commitlint --from origin/main --to HEAD
```

> Also verify every commit in `origin/main..HEAD` has a `Signed-off-by:` trailer **and** a valid GPG signature (`git log --format='%G?' origin/main..HEAD` — every line must be `G`). If any hard policy check fails, stop and report.

Author: Done — replaced the partly-manual "verify Signed-off-by trailer" prose with a single combined verification command that asserts both checks per commit:

```bash
git log --format='%G? %(trailers:key=Signed-off-by,valueonly,separator=%x20) %h %s' origin/main..HEAD
```

Each line must start with `G` (good GPG signature) and have a non-empty `Signed-off-by` value before the SHA. Fail-fast block is now fully actionable instead of relying on manual inspection.

See commit ee7e63f2.

Then run advisory review on:

git diff origin/main...HEAD — read it. Check against CLAUDE.md and .claude/rules/:

- Consider: any browser-only APIs (window, document, etc.) — if found outside isPlatformBrowser guards, flag for the user to confirm (SSR risk)
- Consider: hard-coded brand hex values — if found, suggest using lfxColors scales instead (avoids drift when LF brand updates)
- Module scope respected
- No nested ternaries
- No interface Foo {} inside apps/

Advisory findings are non-blocking — surface them to the user, who decides whether to fix before pushing.

## Step 5: Push

git push -u origin HEAD

If rejected, stop. Do not force-push. Suggest git pull --rebase origin main.

## Step 6: Open PR

Find LFXV2 ticket:

1. Branch name pattern LFXV2-XXXX
2. Commit messages
3. Ask user if neither

PR body template:

- Summary (generated from diff + commits)
- Linked ticket with JIRA URL
- Changes (grouped by module)
- How to test
- Screenshots placeholder
- Pre-merge checklist (rendered into the PR body):
  - [x] `yarn check-types` clean

>   - [x] `yarn lint` clean

Author: Fixed — the checklist labels now match the gates Step 3 actually runs:

- `yarn lint` → `yarn lint:check`
- Added a `yarn format:check` line (Step 3 runs it, but it was missing from the checkbox list)

`yarn check-types`, `yarn test`, and `yarn build` were already aligned.

See commit ee7e63f2.

  - [x] `yarn test` passing
  - [x] `yarn build` passing (SSR safe)
  - [x] DCO sign-off (`--signoff`) and GPG signed (`-S`) on every commit
  - [x] Self-reviewed against `CLAUDE.md` conventions
  - [ ] CodeRabbit review addressed
  - [ ] Code-owner review (required for `lfx-preflight` protected files)
  - [ ] e2e run if UI flow changed

- Footer: 🤖 Drafted via /lfx-pr — Claude Code

gh pr create --base main --title "<first commit subject or branch name>" --body-file <body>

> 

Author: Good catch — the branch-name fallback would produce titles like `feat/LFXV2-1697` (with slash + ticket), which violates the repo's PR-title rules and the PR-title-lint CI check.

Dropped the fallback. The PR title is now derived strictly from the first commit subject, with the requirement that it be a valid conventional-commit title (`type(scope): description`, lowercase, no JIRA ticket). If the first commit subject doesn't satisfy that — e.g., the branch only has merge commits or the subject violates the rules — the command asks the user for a valid title before invoking `gh`.

See commit ee7e63f2.

- git rev-parse --abbrev-ref HEAD → if "main", stop: "Refusing to PR from main."
- git rev-list --count origin/main..HEAD → if 0, stop: "No commits ahead of origin/main."
- git status --porcelain

If git status --porcelain is non-empty:

- If `$ARGUMENTS` includes (or clearly implies) a commit subject — use it and proceed to Step 2.
- Otherwise, propose a commit subject derived from the staged diff and ask the user to confirm or revise before committing.

If working tree is clean, skip Step 2.

## Step 2: Commit (only if dirty)

Validate subject:

- Format: <type>(<scope>): <subject> — Angular convention
- Allowed types: feat, fix, refactor, perf, test, docs, style, build, ci, revert
- Banned: chore:
- Length: ≤72 chars
- Scope: lowercase, describes the affected area (e.g., auth, ui, api, docs, ci, dx) — see `.claude/rules/commit-workflow.md`
- When in doubt, let `yarn commitlint` (run by commit-msg hook) be the source of truth

If invalid, stop and report. Otherwise:

```bash
git add -A
git commit --signoff -S -m "<subject>"
```

> **Note:** Both `--signoff` (DCO) and `-S` (GPG signing) are required by repo policy. Configure GPG if you haven't already.

> 

Reviewer 2: **NIT** — Rule: `.claude/rules/development-rules.md` § Documentation Maintenance (remove duplicates)

The blockquote at line 47 and the prose at line 49 say nearly the same thing back-to-back about DCO+GPG being mandatory. Consider merging into one statement.

Author: Fixed in `ee3c15d9` — merged the blockquote + prose into a single Note. Also added the `.claude/rules/commit-workflow.md` link in the same edit so it satisfies @copilot's open thread too.

> Both `--signoff` (DCO) and `-S` (GPG signing) are mandatory per repo policy. If either fails, stop — do not push unsigned commits.

Reviewer 2: **SHOULD_FIX** — Rule: UNRESOLVED Copilot thread

The reviewer's open thread asks for an in-repo citation for the GPG-required policy. Author replies that PR #674 documents it. Recommend: add an inline link to #674 in the Note so the policy is at least traceable from inside this file while #674 is in flight.

Suggested fix: append to the blockquote:

> See PR #674 which adds this policy to `CLAUDE.md` and `.claude/rules/commit-workflow.md`.

Author: Addressed in `ee3c15d9` via the same edit as @copilot's thread above — the Note now links to `.claude/rules/commit-workflow.md` § Commit Signing, which is the in-repo home of the policy that PR #674 added. The policy is therefore traceable from inside `lfx-pr.md` without needing a PR-number reference.

> Both `--signoff` (DCO) and `-S` (GPG signing) are mandatory per repo policy. If either fails, stop — do not push unsigned commits.

Author: Linked in `ee3c15d9` — the Step-2 Note now points to `.claude/rules/commit-workflow.md` § Commit Signing as the authoritative policy. That file documents both the DCO sign-off and GPG signing requirements with setup instructions, so the requirement is verifiable from inside the repo.

## Step 3: Quality gates (fail-fast)

Run sequentially, stop on first failure, surface output:

```bash
yarn check-types
yarn lint:check
yarn format:check
yarn test
yarn build
```

Gates are read-only (`:check` variants) — they do not mutate the working tree. If lint or format reports issues, fix them, return to Step 2 to commit the fixes, then re-run gates. Do not skip `yarn build` (catches SSR breaks).

## Step 4: Self-review

Run these hard policy checks first (fail-fast):

```bash
./check-headers.sh
yarn commitlint --from origin/main --to HEAD
```

Also verify every commit in `origin/main..HEAD` has both a `Signed-off-by:` trailer and a valid GPG signature:

```bash
git log --format='%G? %(trailers:key=Signed-off-by,valueonly,separator=%x20) %h %s' origin/main..HEAD
```

> Each line must start with `G` (good GPG signature) **and** have a non-empty `Signed-off-by` value before the SHA. If any commit fails either check, stop and report.

Author: Already addressed in the current file — Step 4 explicitly accepts both `G` and `U` (line 79: "Each line must start with `G` or `U` (good signature; `U` means the signing key isn't in the local trust db, which doesn't fail the policy)…"). Marking resolved.

Then run advisory review on:

> git diff origin/main...HEAD — read it. Check against CLAUDE.md and .claude/rules/:

Reviewer 2: **NIT** — Rule: Consistency with surrounding Step 4 fenced blocks

`git diff origin/main...HEAD — read it.` is a bare command in prose, while the two hard-check blocks above are properly fenced. For consistency wrap the diff invocation in a fenced bash block.

```suggestion
```bash
git diff origin/main...HEAD
```

Read the diff and check against `CLAUDE.md` and `.claude/rules/`:
```

Author: Fixed in `ee3c15d9` — `git diff origin/main...HEAD` is now in a fenced bash block with the explanatory prose on a separate line, matching the surrounding hard-check blocks.

- Consider: any browser-only APIs (window, document, etc.) — if found outside isPlatformBrowser guards, flag for the user to confirm (SSR risk)
- Consider: hard-coded brand hex values — if found, suggest using lfxColors scales instead (avoids drift when LF brand updates)
- Module scope respected
- No nested ternaries
- No interface Foo {} inside apps/

Advisory findings are non-blocking — surface them to the user, who decides whether to fix before pushing.

## Step 5: Push

> git push -u origin HEAD

Reviewer 2: **NIT** — Rule: Fenced-block consistency

`git push -u origin HEAD` (Step 5) is bare prose. Every other shell invocation in the file is fenced. Wrap this one too.

```suggestion
```bash
git push -u origin HEAD
```
```

Author: Fixed in `ee3c15d9` — `git push -u origin HEAD` is now fenced, and the rebase suggestion is inline-code (`git pull --rebase origin main`).

If rejected, stop. Do not force-push. Suggest git pull --rebase origin main.

## Step 6: Open PR

Find LFXV2 ticket:

1. Branch name pattern LFXV2-XXXX
2. Commit messages
3. Ask user if neither

PR body template:

- Summary (generated from diff + commits)
- Linked ticket with JIRA URL
- Changes (grouped by module)
- How to test
- Screenshots placeholder
- Pre-merge checklist (rendered into the PR body):
  - [x] `yarn check-types` clean
  - [x] `yarn lint:check` clean
  - [x] `yarn format:check` clean
  - [x] `yarn test` passing
  - [x] `yarn build` passing (SSR safe)
  - [x] DCO sign-off (`--signoff`) and GPG signed (`-S`) on every commit
  - [x] Self-reviewed against `CLAUDE.md` conventions
  - [ ] CodeRabbit review addressed

>   - [ ] Code-owner review (required for `lfx-preflight` protected files)

Reviewer 2: **SHOULD_FIX** — Rule: `.claude/rules/commit-workflow.md` external references; in-repo accuracy

`Code-owner review (required for lfx-preflight protected files)` — there is no skill named `lfx-preflight` in this repo (only `/preflight`). The actual protected-files enforcement lives in `.claude/hooks/guard-protected-files.sh`. Either rename to `/preflight` or reference the hook directly.

```suggestion
  - [ ] Code-owner review (required for files matched by `.claude/hooks/guard-protected-files.sh`)
```

Author: Fixed in `ee3c15d9` — checklist now reads: `Code-owner review (required for files matched by .claude/hooks/guard-protected-files.sh)`. (CLAUDE.md still uses the `lfx-preflight` phrasing — out of scope here, but worth a follow-up.)
